core/pkg/otel/metrics.go

```diff
@@ -0,0 +1,107 @@
+package otel
+
+import (
+	"context"
+	"github.com/prometheus/client_golang/prometheus"
+	"go.opentelemetry.io/otel/metric/unit"
+	"go.opentelemetry.io/otel/sdk/instrumentation"
+	"go.opentelemetry.io/otel/sdk/metric"
+	"go.opentelemetry.io/otel/sdk/metric/aggregation"
+	"time"
+
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric/instrument"
+	semconv "go.opentelemetry.io/otel/semconv/v1.18.0"
+)
+
+type MetricsRecorder struct {
+	httpRequestDurHistogram   instrument.Float64Histogram
+	httpResponseSizeHistogram instrument.Float64Histogram
+	httpRequestsInflight      instrument.Int64UpDownCounter
+	impressions               instrument.Int64Counter
+}
+
+func (r MetricsRecorder) HttpAttributes(svcName, url, method, code string) []attribute.KeyValue {
+	return []attribute.KeyValue{
+		semconv.ServiceNameKey.String(svcName),
+		semconv.HTTPURLKey.String(url),
+		semconv.HTTPMethodKey.String(method),
+		semconv.HTTPStatusCodeKey.String(code),
+	}
+}
+
+func (r MetricsRecorder) HttpRequestDuration(ctx context.Context, duration time.Duration, attrs []attribute.KeyValue) {
+	r.httpRequestDurHistogram.Record(ctx, duration.Seconds(), attrs...)
+}
+
+func (r MetricsRecorder) HttpResponseSize(ctx context.Context, sizeBytes int64, attrs []attribute.KeyValue) {
+	r.httpResponseSizeHistogram.Record(ctx, float64(sizeBytes), attrs...)
+}
+
+func (r MetricsRecorder) InFlightRequestStart(ctx context.Context, attrs []attribute.KeyValue) {
+	r.httpRequestsInflight.Add(ctx, 1, attrs...)
+}
+
+func (r MetricsRecorder) InFlightRequestEnd(ctx context.Context, attrs []attribute.KeyValue) {
+	r.httpRequestsInflight.Add(ctx, -1, attrs...)
+}
+
+func (r MetricsRecorder) Impressions(ctx context.Context, key, variant string) {
+	r.impressions.Add(ctx, 1, []attribute.KeyValue{
+		semconv.FeatureFlagKey(key),
+		semconv.FeatureFlagVariant(variant),
+		semconv.FeatureFlagProviderName("flagd"),
+	}...)
```

Comment:
Seems right!

Reply:
As discussed before, I'll remove this and tackle the introduction of the new metric in a separate PR - this way the changelog is also nicer :)